compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -83,8 +84,18 @@ pub(super) fn extract_refined_covspans<'tcx>(
     // Discard any span that overlaps with a hole.
     discard_spans_overlapping_holes(&mut covspans, &holes);
 
-    // Perform more refinement steps after holes have been dealt with.
+    // Discard spans that overlap in unwanted ways.
     let mut covspans = remove_unwanted_overlapping_spans(covspans);
+
+    // For all empty spans, either enlarge them to be non-empty, or discard them.
+    let source_map = tcx.sess.source_map();
+    covspans.retain_mut(|covspan| {
+        let Some(span) = ensure_non_empty_span(source_map, covspan.span) else { return false };
```

Comment:
After some local testing, it is the `return false` here that causes the issues of 

```
malformed instrumentation profile data: function name is empty
```

So what spans are kept/discarded is changed by this PR. Perhaps due to some of the modifications to the spans that happen right above here?